src/util/__tests__/descendantThoughts.ts

```diff
@@ -0,0 +1,32 @@
+import { descendantThoughts } from '../../util'
+
+it('equal returns true', () => {
+  expect(descendantThoughts([{ value: 'a', rank: 0 }], [{ value: 'a', rank: 0 }])).toBe(true)
+})
+
+it('descendant returns true', () => {
+  expect(descendantThoughts([{ value: 'a', rank: 0 }, { value: 'b', rank: 0 }], [{ value: 'a', rank: 0 }])).toBe(true)
+  expect(descendantThoughts([{ value: 'a', rank: 0 }, { value: 'b', rank: 0 }, { value: 'c', rank: 0 }], [{ value: 'a', rank: 0 }, { value: 'b', rank: 0 }])).toBe(true)
+})
+
+it('subset returns false', () => {
+  expect(descendantThoughts([{ value: 'a', rank: 0 }, { value: 'b', rank: 0 }], [{ value: 'a', rank: 0 }])).toBe(true)
```

Comment:
You can remove the first `expect` from this test since it is part of the "descendant returns true" test.